bin/demo.py

```diff
@@ -2,221 +2,215 @@
-This script is used to demonstrate various features of Jenkins on the DCOS.
+This script is used to demonstrate various features of Jenkins on the DC/OS.
 
 Pre-requisites:
-+ A DCOS CLI is configured and available on the PATH of the host system
-+ A running DCOS cluster greater than version 1.7.0.
++ A running DC/OS cluster greater than version 1.7.0.
 + Python dependencies are installed (pip install -r requirements.txt)
 
-The continuous delivery demo will create a build pipeline that will deploy a Docker
-container to the DCOS Marathon.
+The continuous delivery demo will create a build pipeline that will deploy a
+Docker container to the DC/OS Marathon.
 
 The dynamic slaves demo will create 50 (by default) "freestyle" Jenkins jobs.
 Each of these jobs will appear as a separate Jenkins build, and will randomly
 pass or fail. The duration of each job will be between 120 and 240 seconds.
 """
 
-import json
+import dcos
 import os
 import random
-import requests
-import shutil
+import sys
 
 from docopt import docopt
-from subprocess import call, CalledProcessError, check_output
-from urlparse import urlparse
+from subprocess import call
+from urllib.parse import urlparse
+
+from shakedown import *
 
 def log(message):
-    print "[demo] {}".format(message)
+    print("[demo]: {}".format(message))
 
 def log_and_exit(message):
     log(message)
     exit(1)
 
-def get_auth_wrapper(token_arg):
-    def auth_wrapper(headers):
-        if token_arg is not None and len(token_arg.strip()) > 0:
-            headers['Authorization'] = "token={}".format(token_arg)
-        return headers
-    return auth_wrapper
-
-auth_func = get_auth_wrapper(None)
-
-def check_and_set_token(jenkins_url):
+@contextlib.contextmanager
+def stdchannel_redirected(stdchannel, dest_filename):
     try:
-        global auth_func
-        command = "dcos config show core.dcos_acs_token"
-        token = check_output(command, shell=True).strip('\n')
-        auth_func = get_auth_wrapper(token)
-        r = requests.get(jenkins_url, headers=auth_func({}), verify=False)
-        if r.status_code == 401:
-            log_and_exit("Not authenticated. Please run `dcos auth login` and try again.")
-    except CalledProcessError:
-        log_and_exit ("Not authenticated. Please run `dcos auth login` and try again.")
+        oldstdchannel = os.dup(stdchannel.fileno())
+        dest_file = open(dest_filename, 'w')
+        os.dup2(dest_file.fileno(), stdchannel.fileno())
+        yield
+    finally:
+        if oldstdchannel is not None:
+            os.dup2(oldstdchannel, stdchannel.fileno())
+        if dest_file is not None:
+            dest_file.close()
+
+def check_and_set_token():
+    dcos_username = arguments['--dcos-username']
+    dcos_password = arguments['--dcos-password']
+    try:
+        token = shakedown.authenticate(dcos_username, dcos_password)
+        dcos.config.set_val('core.dcos_acs_token', token)
+    except:
+        log_and_exit('!! authentication failed')
```

Comment:
Can we make this error message more informative? Perhaps something like "`Authentication against DC/OS failed. Please either authenticate using the DC/OS CLI or provide the --dcos-username and --dcos-password options.`"